File: 2023-2024/sourcecode/mod1/lab3/52ViewPager2FragmentStateAdapterJava/app/src/main/java/com/example/a52_viewpager2fragmentstateadapter_java/PagerBasicAdapter.java
package com.example.a52_viewpager2fragmentstateadapter_java;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager.widget.PagerAdapter;
import androidx.viewpager2.adapter.FragmentStateAdapter;

public class PagerBasicAdapter extends FragmentStateAdapter {
    public PagerBasicAdapter(@NonNull FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        return new TemplateFragment(position);
    }

    @Override
    public int getItemCount() {
        return 4;
    }
}

File: 2023-2024/sourcecode/mod1a/lab2/51ViewPager2FragmentStateAdapterKotlin/app/src/main/java/com/example/a51_viewpager2fragmentstateadapter_kotlin/PagerBasicAdapter.kt
package layout

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.example.a51_viewpager2fragmentstateadapter_kotlin.TemplateFragment

class PagerBasicAdapter(fragmentActivity: FragmentActivity): FragmentStateAdapter(fragmentActivity) {
    override fun getItemCount(): Int = 4

    override fun createFragment(position: Int): Fragment = TemplateFragment(position)

}